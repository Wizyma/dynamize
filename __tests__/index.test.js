const getModels = require('../index');

const simpleMock = {
  property: 'jest Test',
  variant: ["A", "B"]
}

const complexMock = {
  property: 'jest Test',
  variantA: ["A", "B"],
  variantB: ["jest", "fb"],
  variantE: ["BA"],
  variantR: [1]
}

const expectedMock = [{
    variantA: 'A',
    variantB: 'jest',
    variantE: 'BA',
    variantR: 1,
    property: 'jest Test'
  },
  {
    variantA: 'A',
    variantB: 'fb',
    variantE: 'BA',
    variantR: 1,
    property: 'jest Test'
  },
  {
    variantA: 'B',
    variantB: 'jest',
    variantE: 'BA',
    variantR: 1,
    property: 'jest Test'
  },
  {
    variantA: 'B',
    variantB: 'fb',
    variantE: 'BA',
    variantR: 1,
    property: 'jest Test'
  }
]

describe('test complex getModels', () => {
  const models = getModels(complexMock);

  it('should match snapshot', () => {
    expect(models).toMatchSnapshot();
  })

  it('length should be equal 4', () => {
    expect(models.length).toBe(4);
  })

  it('should match expected value', () => {
    expect(models).toEqual(expect.arrayContaining(expectedMock))
  })
})

describe('test simple mock getModels', () => {
  const models = getModels(simpleMock);

  it('should match snapshot', () => {
    expect(models).toMatchSnapshot();
  })

  it('length should be equal 2', () => {
    expect(models.length).toBe(2);
  })

  it('should match expected value', () => {
    expect(models).toEqual(expect.arrayContaining([ 
      { variant: 'A', property: 'jest Test' },
      { variant: 'B', property: 'jest Test' } ]
    ))
  })
})
