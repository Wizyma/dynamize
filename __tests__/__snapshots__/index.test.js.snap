// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test complex getModels should match snapshot 1`] = `
Array [
  Object {
    "property": "jest Test",
    "variantA": "A",
    "variantB": "jest",
    "variantE": "BA",
    "variantR": 1,
  },
  Object {
    "property": "jest Test",
    "variantA": "A",
    "variantB": "fb",
    "variantE": "BA",
    "variantR": 1,
  },
  Object {
    "property": "jest Test",
    "variantA": "B",
    "variantB": "jest",
    "variantE": "BA",
    "variantR": 1,
  },
  Object {
    "property": "jest Test",
    "variantA": "B",
    "variantB": "fb",
    "variantE": "BA",
    "variantR": 1,
  },
]
`;

exports[`test simple mock getModels should match snapshot 1`] = `
Array [
  Object {
    "property": "jest Test",
    "variant": "A",
  },
  Object {
    "property": "jest Test",
    "variant": "B",
  },
]
`;
